src/Bicep.Core/Semantics/JsonObjectParser.cs

```diff
@@ -0,0 +1,20 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+using Bicep.Core.Diagnostics;
+using Bicep.Core.Parsing;
+using Microsoft.WindowsAzure.ResourceStack.Common.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Bicep.Core.Semantics
+{
+    public class JsonObjectParser : ObjectParser
+    {
+        /// <summary>
+        /// TryFromJson returns null if the fileContent is not a valid JSON object
+        /// </summary>
+        override protected JToken ExtractTokenFromObject(string fileContent)
```

Comment:
```suggestion
        override protected JToken? ExtractTokenFromObject(string fileContent)
```

The `TryFromJson` method is coming from a library compiled without [nullable reference types](https://learn.microsoft.com/en-us/dotnet/csharp/nullable-references), so the function claims it will return a `JToken` even though it will sometimes return `null`. Within the Bicep codebase, you should be able to assume that a value whose type isn't explicitly marked as nullable will be non-null. `fileContent.TryFromJson<JToken>()` will have an inferred non-nullable type *within* this function, but it's best not to late that spurious lack of nullability escape from this function. 

In general, if you're returning a value produced by a library that doesn't use nullable reference types, you should mark the return type as nullable unless you're sure the value will never be null.

Reply:
I see that this method has a nullable return type in `ObjectParser`, so feel free to ignore this suggestion.